File: android/src/main/java/expo/modules/hellostorage/HelloStorageModule.kt
package expo.modules.hellostorage

import android.content.Context
import android.content.SharedPreferences
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class HelloStorageModule : Module() {
  private val context
    get() = appContext.reactContext ?: throw Exception("React context unavailable")

  private val sharedPreferences: SharedPreferences
    get() = context.getSharedPreferences("HelloStoragePrefs", Context.MODE_PRIVATE)

  override func definition() = ModuleDefinition {
    Name("HelloStorage")

    AsyncFunction("setItem") { key: String, value: String ->
      sharedPreferences.edit().putString(key, value).apply()
    }

    AsyncFunction("getItem") { key: String ->
      return@AsyncFunction sharedPreferences.getString(key, null)
    }
  }
}

File: ios/HelloStorageModule.swift
import ExpoModulesCore

public class HelloStorageModule: Module {
  public func definition() -> ModuleDefinition {
    Name("HelloStorage")

    AsyncFunction("setItem") { (key: String, value: String) in
      UserDefaults.standard.set(value, forKey: key)
    }

    AsyncFunction("getItem") { (key: String) -> String? in
      return UserDefaults.standard.string(forKey: key)
    }
  }
}